spec/system/check_ahoy_tracking_spec.rb

```diff
@@ -2,192 +2,137 @@
-      expect(events.size).to eq 1
+      expect(alpha_event.where(name: 'module_start').size).to eq 1
     end
   end
 
   context 'when a user has visited the first page of the confidence check' do
-    before do
-      view_pages_before_confidence_check(alpha)
-    end
-
     it 'ahoy tracks the starting of the confidence check - track(confidence_check_start)' do
+      view_pages_before_confidence_check(alpha)
       visit '/modules/alpha/content-pages/1-3-3'
       click_on 'Next'
       expect(page).to have_current_path '/modules/alpha/confidence-check/1-3-3-1', ignore_query: true
-      # check ahoy is tracking the event
-      events = Ahoy::Event.where(user_id: user.id, name: 'confidence_check_start').where_properties(training_module_id: 'alpha')
-      expect(events.size).to eq 1
+      expect(alpha_event.where(name: 'confidence_check_start').size).to eq 1
     end
   end
 
   context 'when a user has answered a confidence check question' do
-    before do
+    it 'ahoy tracks when a confidence check question has been answered correctly - track(questionnaire_answer)' do
       view_pages_before_confidence_check(alpha)
       visit '/modules/alpha/confidence-check/1-3-3-1'
       check 'Correct answer 1'
       check 'Correct answer 2'
       click_on 'Next'
-    end
 
-    it 'ahoy tracks when a confidence check question has been answered correctly - track(questionnaire_answer)' do
       expect(page).to have_current_path '/modules/alpha/confidence-check/1-3-3-2', ignore_query: true
-      # check ahoy is tracking the event
-      events = Ahoy::Event.where(user_id: user.id, name: 'questionnaire_answer').where_properties(success: true, training_module_id: 'alpha', type: 'confidence_check')
-      expect(events.size).to eq 1
+      expect(alpha_event.where(name: 'questionnaire_answer').where_properties(success: true, type: 'confidence_check').size).to eq 1
     end
   end
 
   context 'when a user has finished the confidence check (at the thank you page)' do
-    before do
-      view_pages_before_confidence_check(alpha)
-    end
-
     it 'ahoy tracks the completion of the confidence check - track(confidence_check_complete)' do
+      view_pages_before_confidence_check(alpha)
       visit '/modules/alpha/content-pages/1-3-3'
       click_on 'Next'
-      2.times do
-        check 'Correct answer 1'
-        check 'Correct answer 2'
-        click_on 'Next'
-      end
-      choose 'Correct answer 1'
-      click_on 'Next'
+      complete_confidence_check
+
       expect(page).to have_current_path '/modules/alpha/content-pages/1-3-3-4', ignore_query: true
-      # check ahoy is tracking the event
-      events = Ahoy::Event.where(user_id: user.id, name: 'confidence_check_complete').where_properties(training_module_id: 'alpha')
-      expect(events.size).to eq 1
+      expect(alpha_event.where(name: 'confidence_check_complete').size).to eq 1
```

Comment:
HMW get a lazily loaded `module_events` collection for the key we are interested in, is that possible?